relay/http.go

```diff
@@ -247,19 +283,57 @@ func jsonError(w http.ResponseWriter, code int, message string) {
 }
 
 type httpBackend struct {
-	name     string
-	location string
+	name        string
+	location    string
+	retryBuffer *retryBuffer
+	client      *http.Client
+	buffering   int32
 }
 
-func (b *httpBackend) post(buf []byte, query string) (*http.Response, error) {
+func (b *httpBackend) post(buf []byte, query string) (response *http.Response, err error) {
 	req, err := http.NewRequest("POST", b.location, bytes.NewReader(buf))
 	if err != nil {
-		return nil, err
+		return
 	}
 
 	req.URL.RawQuery = query
 	req.Header.Set("Content-Type", "text/plain")
 	req.Header.Set("Content-Length", fmt.Sprint(len(buf)))
 
-	return http.DefaultClient.Do(req)
+	// Check if we are in buffering mode
+	var buffering int32
+	if b.retryBuffer != nil {
```

Comment:
This check can be removed, and probably the local variable. Just do `if atomic.LoadInt32(&b.buffering) == 0 {` as the initial check. If we have no retry buffer, it'll never be anything other than zero, and if we do, we're just adding an extra check for something that isn't going to change for the life of the program.